contracts/swaps/connectors/SwapsImplBalancer_POLYGON.sol

```diff
@@ -109,9 +109,9 @@ contract SwapsImplBalancer_POLYGON is State, ISwapsImpl {
             }
         }
         if (amountOutSpecified > amountOut) {
-            swapParams[0].amount = swapParams[0].amount.sub(amountOutSpecified -amountOut);
+            swapParams[swapParams.length-1].amount = swapParams[swapParams.length-1].amount.sub(amountOutSpecified -amountOut);
         } else if (amountOutSpecified < amountOut) {
-            swapParams[0].amount += amountOut - amountOutSpecified;
+            swapParams[swapParams.length-1].amount += amountOut - amountOutSpecified;
```

Comment:
1) why not add .add( in this case? also += might be overlooked sometime
2) why not use abs(amountOut - amountOutSpecified)
3) this ifelse misses the situation when they are equal `amountOutSpecified == amountOut`

Reply:
1. It cannot overflow since the amountOutSpecified is totaled from amounts and it is the difference so therefore impossible to overflow.
2. abs adds gas costs when they do equal each other since an extra unnecessary write to memory and calculation.
3. When equal nothing should be done as this adjusts the payload to match the desired swap amounts.